app/controllers/api/base_controller/generic.rb

```diff
@@ -94,6 +94,23 @@ def delete_resource(type, id = nil, _data = nil)
         delete_resource_action(klass, type, id)
       end
 
+      def copy_resource(type, id, data = {})
```

Comment:
This should be in `app/controllers/api/orchestration_template_controller.rb` because it is specific to orchestration templates